libs/Database.js

```diff
@@ -1026,47 +1021,51 @@ class Database {
   }
 
   /**
-   * Used by light nodes to cleanup (unneeded) blocks already verified
-   * by witnesses <= @lastVerifiedBlockNumber
-   * @param lastVerifiedBlockNumber
+   * Used by light nodes to cleanup (unneeded) blocks / transactions already verified
+   * by witnesses <= lastVerifiedBlockNumber - blocksToKeep
    * @returns {Promise<void>}
    */
-  async cleanupBlocks(lastVerifiedBlockNumber) {
+  async cleanupLightNode() {
     if (!this.lightNode) {
       return;
     }
-    // await this.chain.deleteMany({ _id: { $lte: lastVerifiedBlockNumber } }, { session: this.session });
+    const params = await this.findOne({ contract: 'witnesses', table: 'params', query: {} });
+    if (params && params.lastVerifiedBlockNumber) {
+      console.log(`cleaning up light node blocks and transactions`);
+      const cleanupUntilBlock = params.lastVerifiedBlockNumber - 1 - this.blocksToKeep;
+      await this.cleanupBlocks(cleanupUntilBlock);
+      await this.cleanupTransactions(cleanupUntilBlock);
+    }
+  }
 
-    await this.chain.deleteMany({ $and: [{ _id: { $gt: 0 } }, { _id: { $lte: lastVerifiedBlockNumber } }] }, { session: this.session });
+  /**
+   * Used by light nodes to cleanup (unneeded) blocks already verified
+   * by witnesses <= lastVerifiedBlockNumber - blocksToKeep
+   * @param cleanupUntilBlock cleanup blocks with a smaller blockNumber
+   * @returns {Promise<void>}
+   */
+  async cleanupBlocks(cleanupUntilBlock) {
+    await this.chain.deleteMany({ $and: [{ _id: { $gt: 0 } }, { _id: { $lte: cleanupUntilBlock } }] }, { session: this.session });
```

Comment:
Should this be >= 0 instead of > 0?  There is a block 0, the genesis block.

Reply:
It was done this way because `createGenesisBlock` in `Blockchain.js` will create the genesis block again if it's not there and syncing will start from block 1 again. So block 0 is currently kept also on light nodes. I can either add a comment why it was done this way or change the `createGenesisBlock` method to exclude light nodes. What do you like better?

Reply:
Okay I see.  That's fine then, no problem with keeping the genesis block for light nodes.  You can just add an explanatory comment.